docs/sources/alerting/unified-alerting/message-templating.md

```diff
@@ -0,0 +1,168 @@
+1. Fill in **Name** and **Content** fields.
+1. Click **Save template** button at the bottom of the page.
+
+**Note** The template name used to reference this template in templating is not the value of the **Name** field, but the parameter to `define` tag in the content. When creating a template you can omit `define` entirely and it will be added automatically with same value as **Name** field. It's recommended to use the same name for `define` and **Name** field to avoid confusion.
+
+
+<img  src="/static/img/docs/alerting/unified/templates-create-8-0.png" width="600px">
+
+### Edit a template
+1. In the Grafana side bar, hover your cursor over the **Alerting** (bell) icon and then click **Contact points**.
+1. Find the template you want to edit in the templates table and click the **pen icon** on the right side.
+1. Make any changes and click **Save template** button at the bottom of the page.
+
+### Delete a template
+1. In the Grafana side bar, hover your cursor over the **Alerting** (bell) icon and then click **Contact points**.
+1. Find the template you want to edit in the templates table and click the **trash can icon** on the right side.
+1. A confirmation dialog will open. Click **Yes, delete**. 
+
+**Note** You will not be prevented from deleting templates that are in use somewhere in contact points or other templates. Be careful!
+
+### Use a template in a contact point field
+
+Simply enter `{{ template "templatename" . }}` into a contact point field, where `templatename` is the `define` parameter of a template. 
+
+<img  src="/static/img/docs/alerting/unified/contact-points-use-template-8-0.png" width="600px">
+
+
+### Template examples
+
+Template to render a single alert:
+```
+{{ define "alert" }}
+  [{{.Status}}] {{ .Labels.alertname }}
+  
+  Labels:
+  {{ range .Labels.SortedPairs }}
+    {{ .Name }}: {{ .Value }}
+  {{ end }}
+  
+  {{ if gt (len .Annotations) 0 }}
+  Annotations:
+  {{ range .Annotations.SortedPairs }}
+    {{ .Name }}: {{ .Value }}
+  {{ end }}
+  {{ end }}
+
+  {{ if gt (len .SilenceURL ) 0 }}
+    Silence alert: {{ .SilenceURL }}
+  {{ end }}
+  {{ if gt (len .DashboardURL ) 0 }}
+    Go to dashboard: {{ .Dashboard URL }}
+  {{ end }}
+{{ end }}
+```
+
+Template to render entire notification message:
+```
+{{ define "message" }}
+  {{ if gt (len .Alerts.Firing) 0 }}
+    {{ len .Alerts.Firing }} firing:
+    {{ range .Alerts.Firing }} {{ template "alert" .}} {{ end }}
+  {{ end }}
+  {{ if gt (len .Alerts.Resolved) 0 }}
+    {{ len .Alerts.Resolved }} resolved:
+    {{ range .Alerts.Resolved }} {{ template "alert" .}} {{ end }}
+  {{ end }}
+{{ end }}
+```
+
+## Managing templates for an external Alertmanager
+
+Grafana alerting UI supports managing external Alertmanager configuration. Add an [Alertmanager datasource]({{< relref "../../datasources/alertmanager.md" >}}) and a dropdown will appear at the top of the page, allowing you to select either `Grafana` or an external Alertmanager. 
```

Comment:
```suggestion
Grafana alerting UI supports managing external Alertmanager configuration. Once you add an [Alertmanager data source]({{< relref "../../datasources/alertmanager.md" >}}), a dropdown displays at the top of the page, allowing you to select either `Grafana` or an external Alertmanager data source. 
```

Reply:
The text above is a good candidate for single sourcing.

Reply:
Could you give an example how to do that?